packages/builder/src/stores/builder/components.ts

```diff
@@ -1100,12 +1126,25 @@ export class ComponentStore extends BudiStore<ComponentState> {
         (setting: ComponentSetting) => setting.key === name
       )
 
+      // Datasource setting changes should only count if the source has been entirely replaced
+      const isDatasource = updatedSetting?.type === "dataSource"
+      const sourceModified = this.isDatasourceUpdated(
+        component,
+        updatedSetting,
+        value
+      )
+
       // Reset dependent fields
       settings.forEach((setting: ComponentSetting) => {
         const needsReset =
           name === setting.resetOn ||
           (Array.isArray(setting.resetOn) && setting.resetOn.includes(name))
+
         if (needsReset) {
+          // Ignore the reset if the updated datasource was not replaced
+          if (isDatasource && !sourceModified) {
+            return
+          }
```

Comment:
NAB: we can hoist this right up to just above where we compute `isDatasource` and `sourceModified is computed, since if this is a datasource setting and there's no change we want to just abort entirely.

In fact, the snippet up above (line 1130) could effectively just be:
```js
if (updatedSetting?.type === "dataSource") {
  if (!this.isDatasourceUpdated(component, updatedSetting, value)) {
    return false
  }
}
```

Just no need to have this logic down in the loop, since if the DS didn't change this should just be a NOP.

Reply:
Yo, thanks for taking a look! I ran through that scenario and if you bail out at that point, the update on line 1169 will be skipped entirely. In this instance we aren't trying to abort the update, just any side-affects of the resetOn if they haven't been met.